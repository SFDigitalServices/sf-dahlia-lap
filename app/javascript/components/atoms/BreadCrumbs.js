import React from 'react'
import PropTypes from 'prop-types'

const Item = ({ item, current }) => {
  if (current) {
    return (
      <li className='current'>
        <a href={item.link} aria-current='page'>
          {item.title}
        </a>
      </li>
    )
  } else {
    return (
      <li>
        <a href={item.link}>{item.title}</a>
      </li>
    )
  }
}

const BreadCrumbs = ({ items }) => {
  if (!items || items.length === 0) return null

  const lastItem = items[items.length - 1]
  return (
    <nav aria-label='breadcrumb'>
      <ol className='breadcrumbs'>
        {items.slice(0, items.length - 1).map((item) => (
          <Item key={item.title} item={item} />
        ))}
        <Item key={lastItem.name} item={lastItem} current />
      </ol>
    </nav>
  )
}

Item.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired
  })
}

BreadCrumbs.propTypes = {
  items: PropTypes.array.isRequired
}

export default BreadCrumbs
