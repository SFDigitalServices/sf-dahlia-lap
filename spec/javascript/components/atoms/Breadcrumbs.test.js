import React from 'react'
import renderer from 'react-test-renderer'
import BreadCrumbs from 'components/atoms/BreadCrumbs'
import sinon from 'sinon'

describe('BreadCrumbs', () => {
  let items = [
    { title: 'Alpha', link: '/alpah' },
    { title: 'Beta', link: '/beta' },
    { title: 'Gama', link: '/gama' },
    { title: 'Pi', link: '/pi' }
  ]

  test('should render succesfully', () => {
    const component = renderer.create(
      <BreadCrumbs items={items} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
