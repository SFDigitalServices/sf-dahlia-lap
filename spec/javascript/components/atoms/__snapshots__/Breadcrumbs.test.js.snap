// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs should render succesfully 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumbs"
  >
    <li>
      <a
        href="/alpah"
      >
        Alpha
      </a>
    </li>
    <li>
      <a
        href="/beta"
      >
        Beta
      </a>
    </li>
    <li>
      <a
        href="/gama"
      >
        Gama
      </a>
    </li>
    <li
      className="current"
    >
      <a
        aria-current="page"
        href="/pi"
      >
        Pi
      </a>
    </li>
  </ol>
</nav>
`;
